tests/test_stem.py

```diff
@@ -585,3 +618,101 @@ def test_set_output_name_new_stage_with_json_output(self, create_default_model:
 
         # check if the output name is set correctly
         assert new_stage.output_settings[0].output_name == "json_output_stage_3"
+
+    def test_finalise_stage_json_output(self):
```

Comment:
i dont think this test is needed, as you are testing the functionalities in test_model.py

Reply:
The test is needed. The test in model checks the raising of errors but the the test in stem checks that the actual json file produced has keys in the right order.